Fix cannot scroll acknowledgement bottom sheet of user list when there are many users

hpkhanh1610 opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A brief description of what this pull request does.
-->
When there are many users in the acknowledgement list, we cannot scroll the list to the bottom.

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket or fixes a reported issue, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-mobile/issues/XXXXX

Otherwise, link the JIRA ticket.
-->
None.
#### Checklist
<!--
Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.
-->
- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: <!-- Device name(s), OS version(s) --> ios

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs/Videos (for both iOS and Android if possible).
-->
Before fix, we cannot scroll the user list. We must grab it outside the list to expand it first:

https://github.com/mattermost/mattermost-mobile/assets/39702132/395b3a2b-b946-4dc6-a75a-629e854866fe

After fix, we can expand and scroll the whole list (similar to facebook's bottom sheet reaction)

https://github.com/mattermost/mattermost-mobile/assets/39702132/c630c49b-b63f-438d-b409-f296faed9390



#### Release Note
<!--
Add a release note for each of the following conditions:

* New features and improvements, including behavioural changes, UI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense. Newlines are stripped.

Example:

```release-note
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```release-note
NONE
```
-->

```release-note
Fix cannot scroll acknowledgement bottom sheet of user list when there are many users
```

enahum commented on app/components/post_list/post/body/acknowledgements/users_list/users_list.tsx:
```diff
@@ -1,8 +1,8 @@
 
-import React, {useCallback, useRef} from 'react';
-import {FlatList} from 'react-native-gesture-handler';
+import {BottomSheetFlatList} from '@gorhom/bottom-sheet';
```
Using this can be a problem when using on tablets as the bottom sheet provider is not registered. Perhaps look into hownit is done for the empji picker or other places

hpkhanh1610 commented:
hi @enahum, I already updated the PR following your suggestion.
I also found out that there is a bug in current MM's code on long press acknowledgement list on tablet:
![Screen Shot 2023-08-08 at 12 04 15](https://github.com/mattermost/mattermost-mobile/assets/39702132/56563d22-a2fb-48d5-bd6d-72369a29d5b9)
I fixed it as well.

And here is the result after fix on tablet:

https://github.com/mattermost/mattermost-mobile/assets/39702132/cb2dccd6-a6df-4c47-83cb-3a32a84a35de

Please re-review.

enahum commented on app/screens/bottom_sheet/index.tsx:
```diff
@@ -191,10 +191,10 @@ const BottomSheet = ({
     if (isTablet) {
         return (
-            <>
+            <GestureHandlerRootView style={styles.container}>
```
this should not be needed, can you check the function `withGestures` in `app/screens/index.tsx` to verify you have the latest main branch?

hpkhanh1610 replied:
Hi, I already just pulled the latest code.

I wonder if you refer to this code:
![Screen Shot 2023-08-08 at 19 44 54](https://github.com/mattermost/mattermost-mobile/assets/39702132/f4542d04-cc06-4f7e-9af0-066637d55df5)

I think the code lacks `withGestures` before `withSafeAreaInsets...`. What do you think if I add `withGestures` here for the bottom sheet, and remove the `GestureHandlerRootView` in this line of code?

enahum replied:
ok so the better fix is to wrap this with the `withGestures` function

larkox reviewed (approved):
LGTM. Nevertheless, I wonder if with this change, we should remove the `GestureHandlerRootView` in https://github.com/mattermost/mattermost-mobile/blob/eb7432dc2722e7316ec07a08231bca3b21f96a58/app/screens/bottom_sheet/index.tsx#L202

@enahum thoughts?

enahum commented:
> LGTM. Nevertheless, I wonder if with this change, we should remove the `GestureHandlerRootView` in
> 
> https://github.com/mattermost/mattermost-mobile/blob/eb7432dc2722e7316ec07a08231bca3b21f96a58/app/screens/bottom_sheet/index.tsx#L202
> 
> @enahum thoughts?

Yes that should not be needed after the change

hpkhanh1610 commented:
> LGTM. Nevertheless, I wonder if with this change, we should remove the `GestureHandlerRootView` in
> 
> https://github.com/mattermost/mattermost-mobile/blob/eb7432dc2722e7316ec07a08231bca3b21f96a58/app/screens/bottom_sheet/index.tsx#L202
> 
> @enahum thoughts?

Yes! So let me remove it totally if you both already agreed.

enahum commented:
> > LGTM. Nevertheless, I wonder if with this change, we should remove the `GestureHandlerRootView` in
> > https://github.com/mattermost/mattermost-mobile/blob/eb7432dc2722e7316ec07a08231bca3b21f96a58/app/screens/bottom_sheet/index.tsx#L202
> > 
> > @enahum thoughts?
> 
> Yes! So let me remove it totally if you both already agreed.

perfect, just make sure to test it to be on the safe side

enahum commented:
@larkox who is the QA person assigned to this?

larkox commented:
@yasserfaraazkhan Please make sure other bottom sheets also work as expected, both in mobile and in tablet.

mattermost-build commented:
Building app in separate branch.

yasserfaraazkhan reviewed (approved):
thank you for the changes. the user list is scrollable on android and iOS

![rn_image_picker_lib_temp_e5aeb1fc-0250-4c26-abb9-e5b0a01fd0f6](https://github.com/mattermost/mattermost-mobile/assets/15979783/e1c3f29f-5687-4ec2-a64e-683a96a2234f)
<img width="696" alt="Screenshot 2023-08-09 at 3 17 42 AM" src="https://github.com/mattermost/mattermost-mobile/assets/15979783/c0ff5ddf-69ce-41ab-98b8-475c92c8e8b9">
